loopdb/protocol_version.go

```diff
@@ -0,0 +1,76 @@
+package loopdb
+
+import (
+	"math"
+
+	"github.com/lightninglabs/loop/looprpc"
+)
+
+// ProtocolVersion represents the protocol version (declared on rpc level) that
+// the client declared to us.
+type ProtocolVersion uint32
+
+const (
+	// ProtocolVersionLegacy indicates that the client is a legacy version
+	// that did not report its protocol version.
+	ProtocolVersionLegacy ProtocolVersion = 0
+
+	// ProtocolVersionMultiLoopOut indicates that the client supports multi
+	// loop out.
+	ProtocolVersionMultiLoopOut ProtocolVersion = 1
+
+	// ProtocolVersionSegwitLoopIn indicates that the client supports segwit
+	// loop in.
+	ProtocolVersionSegwitLoopIn ProtocolVersion = 2
+
+	// ProtocolVersionPreimagePush indicates that the client will push loop
+	// out preimages to the sever to speed up claim.
+	ProtocolVersionPreimagePush ProtocolVersion = 3
+
+	// ProtocolVersionUserExpiryLoopOut indicates that the client will
+	// propose a cltv expiry height for loop out.
+	ProtocolVersionUserExpiryLoopOut ProtocolVersion = 4
+
+	// ProtocolVersionUnrecorded is set for swaps were created before we
+	// started saving protocol version with swaps.
+	ProtocolVersionUnrecorded ProtocolVersion = math.MaxUint32
+
+	// CurrentRpcProtocolVersion defines the version of the RPC protocol
+	// that is currently supported by the loop client.
+	CurrentRPCProtocolVersion = looprpc.ProtocolVersion_USER_EXPIRY_LOOP_OUT
+
+	// CurrentInteranlProtocolVersionInternal defines the RPC current
+	// protocol in the internal representation.
+	CurrentInternalProtocolVersion = ProtocolVersionUserExpiryLoopOut
```

Comment:
Somehow I keep thinking of a hard cast here instead of always updating two fields...

Reply:
agree, fixed